### PR TITLE
Adding vuid 01636

### PR DESCRIPTION
**Pull request by ziga-lunarg**
splintInstanceBindRegionCount in BindImageMemoryDeviceGroupInfo must either be 0 or equal to the number of physical devices in the logical device squared

### CODE REVIEW

**Comment by lunarpapillo** [comment]
The CI failure seems like a sporadic **GitHub** timeout.  I've retriggered.

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -13676,6 +13676,101 @@ TEST_F(VkLayerTest, InvalidImageSplitInstanceBindRegionCount) {
...
+    vk::EnumeratePhysicalDeviceGroups(instance(), &physical_device_group_count, physical_device_group.data());
+    VkDeviceGroupDeviceCreateInfo create_device_pnext = {};
+    create_device_pnext.sType = VK_STRUCTURE_TYPE_DEVICE_GROUP_DEVICE_CREATE_INFO;
+    create_device_pnext.physicalDeviceCount = physical_device_group[0].physicalDeviceCount;
+    create_device_pnext.pPhysicalDevices = physical_device_group[0].physicalDevices;
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &create_device_pnext));
+
+    PFN_vkBindImageMemory2KHR vkBindImageMemory2Function = nullptr;
+
+    if (bind_memory_2_extension) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:13715]: This must be true at this point based on line 13687 above, no?

**Review by ncesario-lunarg — COMMENTED** [review]
I think this looks good, though I'm a little confused by some conditional refactoring.

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:13715]: True.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -12443,15 +12443,31 @@ bool CoreChecks::ValidateBindImageMemory(uint32_t bindInfoCount, const VkBindIma
...
+            if (bind_image_memory_device_group_info) {
+                if (bind_image_memory_device_group_info->splitInstanceBindRegionCount != 0) {
+                    if (!(image_state->createInfo.flags & VK_IMAGE_CREATE_SPLIT_INSTANCE_BIND_REGIONS_BIT)) {
+                        skip |= LogError(bind_info.image, "VUID-VkBindImageMemoryInfo-pNext-01627",
+                                         "%s: pNext of VkBindImageMemoryInfo contains VkBindImageMemoryDeviceGroupInfo with "
+                                         "splitInstanceBindRegionCount (%" PRIi32
+                                         ") not equal to 0 and %s is not created with "
+                                         "VK_IMAGE_CREATE_SPLIT_INSTANCE_BIND_REGIONS_BIT.",
+                                         error_prefix, bind_image_memory_device_group_info->splitInstanceBindRegionCount,
+                                         report_data->FormatHandle(image_state->image()).c_str());
```
[the remark this answers is not in the document]
Comment (ziga-lunarg) [reply layers/core_validation.cpp]: Yes, you are correct, fixed.

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**